src/esn.calendar.libs/app/services/calendar-api.js

```diff
@@ -302,5 +304,15 @@ require('./http-response-handler.js');
       return calDavRequest('put', eventPath, headers, body)
         .then(calHttpResponseHandler([200, 204]));
     }
+
+    /**
+     * GET request used to download the calendar ICS file
+     * @param {String}    calendarPath the calendar ics url to download from.
+     * @param {Function}  handler
+     */
+    function exportCalendar(calendarPath, handler) {
+      calDavRequest('get', calendarPath, { Accept: CAL_ACCEPT_EXPORT_HEADER })
+        .then(calHttpResponseHandler(200, handler));
+    }
```

Comment:
You should only pass calendarHomeId & calendarId, and this function should handle building the whole URL.

And I think we should use a Promise instead of the old, dreaded callback.

Reply:
done.